src/openms/source/CHEMISTRY/TheoreticalSpectrumGenerator.cpp

```diff
@@ -216,6 +216,29 @@ namespace OpenMS
     if (ion_names_dynamic) delete ion_names;
 
     if (sort_by_position_) spectrum.sortByPositionPresorted(chunks.getChunks());
+
+    // set MS Level
+    spectrum.setMSLevel(2);
+
+    // set spectrum type
+    spectrum.setType(MSSpectrum::SpectrumSettings::CENTROID);
+
+    // set precursor
+    Precursor prec;
+
+    if (precursor_charge == 0)
+    {
+      precursor_charge = max_charge +1;
+    }
+    
+    if (precursor_charge <= max_charge)
```

Comment:
this is not correct, IMHO.
In MALDI, all charges are 1.

So, it should be `if (precursor_charge < max_charge)` 
Please also add a 
`@throw Exception::InvalidParameter if precursor_charge < max_charge

Reply:
good point